fix-commonjs-dist.js

```diff
@@ -0,0 +1,3 @@
+import * as fs from 'fs'
+
+fs.writeFileSync('./dist/main/package.json', '{"type": "commonjs"}')
```

Comment:
Node looks at `type` in the nearest `package.json` to tell if a `.js`file should be interpreted as an ESM or commonjs module. You can use `.mjs` or `.cjs` to specify this on a per-file basis, but typescript doesn't have any support for that. See https://github.com/azu/tsconfig-to-dual-package#motivation for more details and links to discussion about this.

I elected to put `type: module` in the top-level package, since my impression is that the ecosystem is *slowly* moving towards ESM style modules.